### PR TITLE
NPSP: Security update for NPSP Settings Relationships Relationships page

### PR DESCRIPTION
**Pull request by psadie-sfdo**
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata
- Classes
-- STG_PanelRel_CTRL_TEST
- Custom Labels
-- commonPermissionErrorMessage
# Deleted Metadata

### CODE REVIEW

File: src/pages/STG_PanelRel.page [hunk 1]
```diff
@@ -1,77 +1,121 @@
...
+                <apex:commandButton id="saveRel" value="{!$Label.stgBtnSave}" status="statusLoad" action="{!saveSettings}" rendered="{!isEditMode}"
+                    immediate="false" rerender="form" styleClass="slds-button slds-button_small slds-button_brand" />
+                <apex:commandButton id="cancelRel" value="{!$Label.stgBtnCancel}" status="statusLoad" action="{!cancelEdit}" rendered="{!isEditMode}"
+                    immediate="true" rerender="form" styleClass="slds-button slds-button_small slds-button_neutral" />
+            </div>
+        </apex:outputPanel>
     </apex:form>
+
+    <script>
+        const doesNotHaveAccess = {!hasAccess} === false;
```
Comment (voduyemi) [inline src/pages/STG_PanelRel.page:103]: @psadie-sfdo What do you think about using `{!NOT(hasAccess)}` instead of `{!hasAccess} === false;` here?

Comment (psadie-sfdo) [reply src/pages/STG_PanelRel.page]: Sounds good to me! updated

**Comment by psadie-sfdo** [comment]
**lurch: attach W-041511

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007haSOQAY" target="_blank">W-041511</a>

File: src/classes/STG_PanelRel_CTRL.cls [hunk 2]
```diff
@@ -39,7 +39,27 @@ public with sharing class STG_PanelRel_CTRL extends STG_Panel {
...
     */
     public override string idPanel() { return 'idPanelRel'; }
-    
+
+    /*********************************************************************************************************
+    * @description NPSP namespace: an empty string if unmanaged, or 'npsp' if managed
+    */
+    public String getNamespace() {
+        String namespace = UTIL_Namespace.getNamespace();
+        if (namespace == null || namespace == '') return 'c';
```
Comment (vmartinov-sforg) [inline src/classes/STG_PanelRel_CTRL.cls:48]:
```suggestion
        if (String.isBlank(namespace)) return 'c';
```

File: src/classes/STG_PanelRel_CTRL.cls [hunk 3]
```diff
@@ -39,7 +39,27 @@ public with sharing class STG_PanelRel_CTRL extends STG_Panel {
...
+        return namespace;
+    }
+
+    /*********************************************************************************************************
+    * @description Returns true if user has full CRUD access to Relationship_Sync_Excluded_Fields__c
+    */
+    public Boolean getHasAccess() {
+        String relationshipObjectName = Schema.SObjectType.Relationship_Sync_Excluded_Fields__c.getName();
+        return UTIL_Permissions.canRead(relationshipObjectName) &&
```
Comment (vmartinov-sforg) [inline src/classes/STG_PanelRel_CTRL.cls:57]: Do we want to throw an exception on canRead()?

Comment (psadie-sfdo) [reply src/classes/STG_PanelRel_CTRL.cls:57]: Good catch, no we do not. Updating!

**Comment by voduyemi** [comment]
**lurch: attach W-041506

**Comment by psadie-sfdo** [comment]
**lurch: detach

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007haRzQAI" target="_blank">W-041506</a>

**Comment by LurchTheButler** [comment]
Detached <a href="https://foundation.my.salesforce.com/a2x1E000007haSOQAY" target="_blank">W-041511</a>

**Pull request merged**